### PR TITLE
Jenkins token auth

### PR DESCRIPTION
**Pull request by tabladrum**
Changes to handle Cloudbees Jenkins Ops Center username/token based auth.
Added more build collector item clean up method: Delete build collector items for jenkins instances that are moved/no longer configured.

### CODE REVIEW

File: jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java [hunk 1]
```diff
@@ -282,10 +297,16 @@ private HttpHeaders createHeaders(final String userId, final String password) {
     }
 
     private String getLog(String buildUrl) {
-        ResponseEntity<String> responseEntity = makeRestCall(
-                URI.create(buildUrl + "consoleText"));
-        String returnJSON = responseEntity.getBody();
+        ResponseEntity<String> responseEntity = null;
+
+        try {
+            responseEntity = makeRestCall(
```
Comment (jdamick) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java]: no need to even create responseEntity, just return the body.

File: jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/HudsonCollectorTask.java [hunk 2]
```diff
@@ -1,213 +1,212 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.scheduling.TaskScheduler;
 import org.springframework.stereotype.Component;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (jdamick) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/HudsonCollectorTask.java]: dont collapse these

File: jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/HudsonCollectorTask.java [hunk 3]
```diff
@@ -1,213 +1,212 @@
...
-import com.capitalone.dashboard.model.HudsonJob;
-import com.capitalone.dashboard.repository.BaseCollectorRepository;
-import com.capitalone.dashboard.repository.BuildRepository;
-import com.capitalone.dashboard.repository.ComponentRepository;
-import com.capitalone.dashboard.repository.HudsonCollectorRepository;
-import com.capitalone.dashboard.repository.HudsonJobRepository;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
+import com.capitalone.dashboard.model.*;
+import com.capitalone.dashboard.repository.*;
```
Comment (jdamick) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/HudsonCollectorTask.java]: dont collapse these

File: jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java (hunk 1 above)
Comment (tabladrum) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java]: yep.

**Comment by jdamick** [comment]
these changes didnt break any unit tests?

**Pull request merged**